compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/JvmInitsGen.java

```diff
@@ -0,0 +1,318 @@
+        // Using object return type since this is similar to a ballerina function without a return.
+        // A ballerina function with no returns is equivalent to a function with nil-return.
+        MethodVisitor mv = cw.visitMethod(Opcodes.ACC_PUBLIC + ACC_STATIC, CURRENT_MODULE_INIT,
+                                          String.format("(L%s;)L%s;", STRAND_CLASS, OBJECT), null, null);
+        mv.visitCode();
+
+        mv.visitMethodInsn(INVOKESTATIC, typeOwnerClass, CREATE_TYPES_METHOD, "()V", false);
+        mv.visitTypeInsn(NEW, typeOwnerClass);
+        mv.visitInsn(DUP);
+        mv.visitMethodInsn(INVOKESPECIAL, typeOwnerClass, JVM_INIT_METHOD, "()V", false);
+        mv.visitVarInsn(ASTORE, 1);
+        mv.visitLdcInsn(module.org.value);
+        mv.visitLdcInsn(module.name.value);
+        mv.visitLdcInsn(module.version.value);
+        mv.visitVarInsn(ALOAD, 1);
+        mv.visitMethodInsn(INVOKESTATIC, String.format("%s", VALUE_CREATOR), "addValueCreator",
+                           String.format("(L%s;L%s;L%s;L%s;)V", STRING_VALUE, STRING_VALUE, STRING_VALUE,
+                                         VALUE_CREATOR),
+                           false);
+
+        // Add a nil-return
+        mv.visitInsn(ACONST_NULL);
+        JvmMethodGenUtils.visitReturn(mv);
+    }
+
+    public void addInitAndTypeInitInstructions(BIRNode.BIRPackage pkg, BIRNode.BIRFunction func) {
+        List<BIRNode.BIRBasicBlock> basicBlocks = new ArrayList<>();
+        nextId = -1;
+        BIRNode.BIRBasicBlock nextBB = new BIRNode.BIRBasicBlock(getNextBBId());
+        basicBlocks.add(nextBB);
+
+        PackageID modID = JvmMethodGenUtils.packageToModuleId(pkg);
+
+        BIRNode.BIRBasicBlock typeOwnerCreateBB = new BIRNode.BIRBasicBlock(getNextBBId());
+        basicBlocks.add(typeOwnerCreateBB);
+
+        nextBB.terminator = new BIRTerminator.Call(null, InstructionKind.CALL, false, modID,
+                                                   new Name(CURRENT_MODULE_INIT),
+                                                   new ArrayList<>(), null, typeOwnerCreateBB, Collections.emptyList(),
+                                                   Collections.emptySet());
+
+        if (func.basicBlocks.size() == 0) {
+            typeOwnerCreateBB.terminator = new BIRTerminator.Return(func.pos);
+            func.basicBlocks = basicBlocks;
+            return;
+        }
+
+        typeOwnerCreateBB.terminator = new BIRTerminator.GOTO(null, func.basicBlocks.get(0));
+
+        basicBlocks.addAll(func.basicBlocks);
+        func.basicBlocks = basicBlocks;
+    }
+
+    public void enrichPkgWithInitializers(Map<String, JavaClass> jvmClassMap, String typeOwnerClass,
+                                          BIRNode.BIRPackage pkg, List<PackageID> moduleImports) {
+        JavaClass javaClass = jvmClassMap.get(typeOwnerClass);
+        BIRNode.BIRFunction initFunc = generateDepModInit(moduleImports, pkg, MODULE_INIT,
```

Comment:
"generateDepModInit" name is bit confusing. AFAIU this generating default methods like start, init